### DISCUSSION TITLE
VM Stopping issue

### DISCUSSION THREAD

**Discussion by sivagithub25** [tomasrudh/AutoShutdownSchedule#16] [Q&A] [answered]
Hi,  My VM has the tag AutoShutdownShedule marked as 2:00. My intention was for the runbook to shutdown the VM at 2:00 AM, but the 2:00 AM scheduled run DID NOT shutdown the VM.
The log shows
<<
[vmname]: Found direct VM schedule tag with value: 2:00
[vmname]: Current time [11/04/2022 02:00:26] falls outside the scheduled shutdown range [2:00], single time so not changing the state.
>>>
If I change the Tag to 13:00, then the shutdown happens without any issue, during the 1:00 PM execution 
[vmname]: Found direct VM schedule tag with value: 13:00
[vmname]: Current time [11/04/2022 13:00:41] falls within the scheduled shutdown range [13:00]
[vmname]: Stopping VM
The parameters used for the job execution are the subscription name and the timestamp value as "AUS Eastern Standard Time".
Can I know if someone else is facing similar issue?
Cheers,
Siva

**Comment by tomasrudh** [comment]
Hi, thanks for letting me know, I will investigate and get back.

/Tomas

**Reply by sivagithub25** [reply]
Thanks a lot. On further testing, I see that the 10:00 AM AEDT scheduled run considers the VM' tag (10:00 AM) to be out of range where as  when we change the tag to 11:00 AM , then the scheduled run at 11:00 AM picks it up correctly. This is something to do with the daylight saving' timezone handling I believe.
I tried placing some debug messages and found that the control does not go inside of the foreach loop during the 10:00 AM run, but it does for the 11:00 AM run.

foreach ($entry in $timeRangeList) {
            if ((CheckScheduleEntry -TimeRange $entry) -eq $true) {
                $scheduleMatched = $true
                $matchedSchedule = $entry
		break
            }
	}

**Comment by tomasrudh** [comment]
Thanks for taking the time to troubleshoot. I think I understand now what is going on, when the script translates AEDT to UTC does it end up on the correct time but one day off. This only happens for some times, depending on if the translation spills over midnight. Give me a day or two and I will fix that.

Thanks again!
/Tomas

**Comment by sivagithub25** [comment]
Thanks Tomas.

Will wait for the fix.

Appreciate your help.

Cheers,
Siva

**Comment by tomasrudh** [comment, marked as answer]
Please get 3,9,1 and see if the issue is solved.

**Reply by sivagithub25** [reply]
Thanks a lot Tom. The new version seem to be working. Appreciate your quick response. Will test for few more iterations and will let you know incase of any issues.
